src/com/facebook/buck/go/GoCompile.java

```diff
@@ -108,15 +113,27 @@ public GoCompile(
     ImmutableList.Builder<Path> compileSrcListBuilder = ImmutableList.builder();
     ImmutableList.Builder<Path> headerSrcListBuilder = ImmutableList.builder();
     ImmutableList.Builder<Path> asmSrcListBuilder = ImmutableList.builder();
+    List<Path> srcFiles = new ArrayList<>();
```

Comment:
This method is already too big, could you move the logic responsible for collecting source files into a standalone method?